library/init/meta/interactive.lean

```diff
@@ -6,8 +6,11 @@ Authors: Leonardo de Moura
 prelude
 import init.meta.tactic init.meta.rewrite_tactic init.meta.simp_tactic
 import init.meta.smt.congruence_closure init.category.combinators
+import init.meta.lean_parser
 
 namespace interactive
+@[reducible] meta def parse {α : Type} (p : lean_parser α) : Type := α
```

Comment:
> While strong typing might just work out for the custom tactic syntax, it will definitely be limiting for the parser extensions.

Sure, the token-initiated parsers should always have type `lean.parser pexpr`; no need for strong typing there.

I'm not sure if I've fully understood your proposal yet. Would you change all the primitive parsers and combinators to `parser pexpr`, or only the top-most parser? In the latter case, what would be the difference to `parse` with `[has_quote]`?